### ISSUE TITLE
Better support of Vim9?

### ISSUE THREAD

**Issue by ajukraine** [vigoux/tree-sitter-viml#118]
Actually I don't know if it's supposed to support Vim9. This is what I get with it.

Comments are not deteced.
Some things like 'def' keyword probably are not highlighted as they should (I dunno?).

<img width="1080" alt="image" src="https://user-images.githubusercontent.com/828545/186735448-608c5119-7b40-4175-a719-6b0d6ed7dc72.png">

<img width="1018" alt="image" src="https://user-images.githubusercontent.com/828545/186735790-670ab945-cc3f-4d60-8a6d-2d062c61976c.png">

**Comment by jchros** [comment]
I think that the only way to parse Vim9 script with tree-sitter is to create a separate grammar file that would be injected on Vimscript files with injection queries that could look something like this:

```
(source_file
  (vim9script_command) (_)* @vim9script))
(compiled_function_definition) @vim9script
(vim9_command (chunk) @vim9script)
```

The reason why is that there are commands that don't do the same things in Vim9 script and in good old Vimscript, and I don't think there is a reliable way to figure out which syntax is supposed to be used except if we take the approach I have described above.  For example, in good old Vimscript, `:#` is an alias for the `:number` command (unless when followed by a `!`), whereas it makes a comment in Vim9 script.

**Comment by clason** [comment]
Sorry, Type9script is for all intents and purposes a different language, so requires a different parser, so this is out of the scope of this project.

Injections are possible (for a limited subset of inclusions) but a matter of writing the right queries; you can do that yourself -- I don't see Neovim maintainers being interested in doing that work for something that is Vim only.

(I would also be interested in why you want tree-sitter support for vim9script given that Vim seems to have no interest in tree-sitter...)

**Comment by vigoux** [comment]
TLDR: As mentionned by the others, we will not add (not even plan to) add Vim9Script to _this_ parser.

As a `neovim` user myself, and as the goal of this parser is to be used in `neovim`, I will not create another parser for Vim9Script.

First of all the amount of work, and the performance impact that would follow from adding Vim9Script is just enormous. Furthermore, the amount of conflicts (within the grammar) that would follow this change is also enourmous. Finally, on a more personnal note, I consider Vim9Script a design mistake, and I will not do anything to allow its propagation into my environment.

Thanks for the report, and if you need help writing a Vim9Script parser, feel free to contact us for guidelines.
